packages/trees/frontend/sirius-components-trees/src/views/TreeView.types.ts

```diff
@@ -112,3 +94,24 @@ export interface GQLGetExpandAllTreePathViewer {
 export interface GQLGetExpandAllTreePathEditingContext {
   expandAllTreePath: GQLTreePath;
 }
+
+export interface GQLTreeEventPayload {
+  __typename: string;
+}
+
+export interface GQLTreeRefreshedEventPayload extends GQLTreeEventPayload {
+  id: string;
+  tree: GQLTree;
+}
+
+export interface UseTreeSubscriptionValue {
+  loading: boolean;
+  tree: GQLTree | null;
+  complete: boolean;
+}
+
+export interface UseTreeSubscriptionState {
+  id: string;
+  tree: GQLTree | null;
+  complete: boolean;
+}
```

Comment:
These are not the concern of sirius-components-trees it's not used in this npm package.

By putting this code here, you are trying to avoid having two almost identical copies of those types in `useModelBrowserSubscription` and `useExplorerSubscription`. Unless we have some maintainability issue with the code, some duplicated code is not an issue I care about. On the other hand, useless code (for `sirius-components-trees`) and useless coupling are both of the most critical issues that I'm trying to avoid.

Removing duplicated code always comes with one critical tradeoff, creating additional coupling.

Now both the lifecycle the explorer and the model browser are tied together with those types. What if someone wants to change one of those? Their first instinct will be to update `UseTreeSubscriptionState` and we will end up with useless code for the other one.

The state of `useExplorerSubscription` needs to fulfill the expectations of `useExplorersubscription` and the state of  `useModelBrowserSubscription` needs to fulfill the expectations of `useModelBrowserSubscription`. It just happens that today both have the same expectations.